crdt/rlist.go

```diff
@@ -926,20 +953,109 @@ func (l *RList) DeleteCharAt(i int) error {
 	return l.DeleteChar()
 }
 
+// +-----------------------------------+
+// | Operations - Insert str container |
+// +-----------------------------------+
+
+//Inserts a string container as a child of the root atom.
+type InsertStr struct{}
+
+func (v InsertStr) AtomPriority() int { return 30 }
+func (v InsertStr) MarshalJSON() ([]byte, error) {
+	return json.Marshal("insert str container")
+}
+
+func (v InsertStr) String() string { return "STR: " }
+
+func (v InsertStr) ValidateChild(child AtomValue) error {
+	switch child.(type) {
+	case InsertChar:
+		return nil
+	case Delete:
+		return nil
+	default:
+		return fmt.Errorf("invalid atom value after InsertStr: %T (%v)", child, child)
+	}
+}
+
+// InsertStr inserts a Str container after the cursor position and advances the cursor.
+func (l *RList) InsertStr() error {
```

Comment:
To avoid copy-pasting code from `addAtom`, we can look for ways to make it share some other functions with `InsertStr`. 

The main difference in this case is that `addAtom` reads position information from the cursor, while InsertStr (currently) always adds the atom as a child of the root node. 

Upon reviewing the `addAtom` function, I believe it'd work just the same if `l.Cursor = Atom{}`, so this function can become simply

```go
func (l *RList) InsertStr() error {
  l.Cursor = Atom{}
  atomID, err := l.addAtom(InsertStr{})
  return err
}
```

The cursor is adjusted within `addAtom` to point to the newly inserted Str.

Reply:
OK